internal/pkg/pkg.go

```diff
@@ -382,3 +382,51 @@ func functionConfigFilePaths(pl *kptfilev1alpha2.Pipeline) (fnConfigPaths sets.S
 	}
 	return fnConfigPaths
 }
+
+// FunctionConfigFilePaths returns a set of config file paths that used by
```

Comment:
How are function config files identified ? By going through the pipeline and extracting `configPath` strings ? Better to explain it here.

Reply:
The logic is defined in `fnConfigFilePaths`. And I think we shouldn't add implementation details in the function comment.

Reply:
What if there is no Kptfile at all and fn-config is passed as an input flag? Will that file still be excluded ?

Reply:
If the directory has no Kptfile it will not be considered as a package so we don't need to do anything special here.

The function config that passed in by command line flag is not considered here as we discussed before. If the config file is not in the package then it will not be included and that's expected. If the config file is in the package, users need to add the file to ignore list if they don't want to include it.

Reply:
I think `eval` should work even if there is no Kptfile.

For example if a directory has one resource file and a fn-config file

`kpt fn eval --image gcr.io/kpt-fn/set-annotation --fn-config my-config.yaml -- foo=bar` should set annotation only on the resource and not on my-config.yaml. Is that the case with this PR ?

Reply:
No that's not the case. We considered this scenario but it doesn't make sense because the `eval` can be called in a chain. For example: `kpt fn eval ... --fn-config my-config.yaml | kpt fn eval ... --fn-config another-config.yaml`. It's impossible to know `another-config.yaml` is a function config in the first `eval` command. So the discussion result is what I described above.

Reply:
This PR is only for ignoring pipeline function config files. For `eval` functions it shouldn't be in this PR even we decide to support it.